### PR TITLE
support for paypal preapproval on website

### PR DESCRIPTION
**Pull request by asood123**
If there is no valid paypal preapproval:
image

when there is one: 
image

### CODE REVIEW

File: frontend/src/actions/users/confirm_preapproval_key.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import { post } from '../../lib/api';
+import * as constants from '../../constants/users';
+
+/**
+ * Confirm approval key
+ */
+
```
Comment (xdamman) [inline frontend/src/actions/users/confirm_preapproval_key.js]: we usually don't add an extra space between the comment of a function and the function.

File: frontend/src/actions/users/get_approval_key.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import * as constants from '../../constants/users';
+
+/**
+ * Get approval key for paypal
+ */
+
+export default (userid, options={}) => {
+  // Redirect to homepage that will confirm the preapproval key
+  const callback = `${options.baseUrl}/?paypalApprovalStatus=`;
+  const preapprovalTemplate = '${preapprovalKey}';
```
Comment (xdamman) [inline frontend/src/actions/users/get_approval_key.js]: Wrong apostrophe. Should use `

Comment (xdamman) [inline frontend/src/actions/users/get_approval_key.js]: But why not just `const preapprovalTemplate = preapprovalKey;` ? Need to force it to be a string?

File: frontend/src/containers/Ledger.js [hunk 3]
```diff
@@ -138,6 +150,15 @@ export class Ledger extends Component {
 
         { this.props.route.type === 'expenses' && <div className='Ledger-container padding40 expenses-container'>
             <div className='line1'>unpaid expenses</div>
+
+            {isHost &&
```
Comment (xdamman) [inline frontend/src/containers/Ledger.js]: shouldn't this be `isHost && !paypalCard` ?
I'd rather have that logic outside of the `<PaypalReminder />` component

File: frontend/src/actions/users/get_approval_key.js (hunk 2 above)
Comment (asood123) [inline frontend/src/actions/users/get_approval_key.js]: We don't know the preapproval key at this point in the sequence, paypal fills it out and it needs that format.

File: frontend/src/containers/Ledger.js (hunk 3 above)
Comment (asood123) [inline frontend/src/containers/Ledger.js]: PaypalReminder needs to decide which message to show and for that it needs the card. It also needs info on details on the card - for example email and eventually a message to expire the card. 

I think it's easier to have a component that's passed a paypalCard and it decides how to display it. We will need to show it other places as well.

**Comment by xdamman** [comment]
ok

**Pull request merged**